pkg/mcpserver/server.go

```diff
@@ -65,30 +64,35 @@ func RunServers(ctx context.Context, mcpFilePath string) error {
 		}
 	}
 
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+	errChn := make(chan error, 1)
+
 	var wg sync.WaitGroup
-	errChn := make(chan error, len(mcpConfig.Servers))
 	wg.Add(len(mcpConfig.Servers))
 
 	for _, s := range mcpConfig.Servers {
 		go func(server *mcpfile.MCPServer) {
 			defer wg.Done()
 			err := RunServer(ctx, server)
 			if err != nil {
-				errChn <- err
+				select {
+				case errChn <- fmt.Errorf("error running server: %s", err.Error()):
+				default:
+				}
 			}
 		}(s)
 	}
 
-	go func() {
-		wg.Wait()
-		close(errChn)
-	}()
-
-	for err := range errChn {
-		log.Printf("error running server: %s", err.Error())
+	var firstErr error
+	select {
+	case firstErr = <-errChn:
+		cancel()
```

Comment:
@Manaswa-S I think we still need to log the error here, otherwise it won't be clear to users why this failed